lib/appsignal/plug.ex

```diff
@@ -78,6 +78,41 @@ if Appsignal.plug? do
       "#{method} #{path}"
     end
 
+    def extract_sample_data(%Plug.Conn{params: params, host: host,
+      method: method, script_name: script_name, request_path: request_path,
+      port: port, query_string: query_string} = conn) do
+
+      %{
+        "params" => Appsignal.Utils.ParamsFilter.filter_values(params),
+        "environment" => %{
+          "host" => host,
+          "method" => method,
+          "script_name" => script_name,
+          "request_path" => request_path,
+          "port" => port,
+          "query_string" => query_string,
+          "request_uri" => url(conn),
+          "peer" => peer(conn)
+        } |> Map.merge(extract_request_headers(conn))
+      }
+    end
+
+    def extract_request_headers(%Plug.Conn{req_headers: req_headers}) do
+      keys = ~w(
+        accept accept-charset accept-encoding accept-language cache-control
+        connection user-agent from referer range
+      )
```

Comment:
I think most x-headers are more common outside of development mode. Such as behind a loadbalancer (HTTP_X_FORWARDED_FOR), nginx (HTTP_X_QUEUE_TIME, HTTP_X_APPLICATION_START), Heroku (HTTP_X_HEROKU_QUEUE_WAIT_TIME), etc.

We can leave some out, but I think some other headers might still be useful for our users to get a better idea why a request errored out (missing header) or why it performed badly (long queue time).

Also, now that I look at it, it doesn't appear like we set the queue time for a request anywhere right now, do we? There's a`set_queue_start` function on Transaction, but I don't see that being called.